docs/src/userguide/loading_iris_cubes.rst

```diff
@@ -215,6 +215,14 @@ constraint to ``load``::
     level_10 = iris.Constraint(model_level_number=10)
     cubes = iris.load(filename, forecast_6 & level_10)
 
+.. note::
+
+    Whilst ``&`` is supported, the ``|`` that might reasonably be expected
+    is not. This is because "or" constraints could lead to a cube which did
+    not cover a hyper-rectangular region if these constraints were on
+    different coordinates. If an "or" type constraint on a single coordinate
+    is useful then it may be obtained by passing a function to :class:`iris.Constraint`.
+
```

Comment:
I think it might also be worth explaining that you can select for for A and B constraints separately, using iris.load_cubes and CubeList.extract_cubes.
E.G. `my_cubes = iris.load_cubes(sourcefile, [cons_sel_A. cons_sel_B, cons_sel_C])`
or, (often) maybe : `cube_A, cube_B, cube_C = iris.load_cubes(sourcefile, [cons_sel_A. cons_sel_B, cons_sel_C])`

So, this is really _not_ an 'OR', but it is similar as long as your separate selections each make a single cube.

Is this any use ??

Reply:
Yes, that's helpful as another place people might want to end up from the initial query that starts them off here. I'll build a reference to that into the final resolution.

Reply:
Rather than adding an example specifically to cover this, I included a link to the load_cubes method with a description of the use case, which I think is the concise but helpful way to go.